### PR TITLE
Finish ExecutionReport implementation to get std out/err stream

### PR DESCRIPTION
**Pull request by mathbagu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
N/A

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
The `getStdOut` and `getStdErr` are not implemented in the only `ExecutionReport` subclass we have in powsybl-core. The methods are implemented in powsybl-afs, for slurm computation.

**What is the new behavior (if this is a feature change)?**
The `DefaultExecutionReport` override the two methods. As this class need to access to the working directory, the `ok` method is removed. It was used only in unit test, so it should be acceptable.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java [hunk 1]
```diff
@@ -22,29 +27,22 @@
...
-        return OK;
-    }
-
     /**
      * Create an execution report with no execution error.
      */
-    public DefaultExecutionReport() {
-        this(Collections.emptyList());
+    public DefaultExecutionReport(Path workingDirectory) {
+        this(workingDirectory, Collections.emptyList());
```
Comment (MioRtia) [inline computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java:38]: Is it relevant to make a constructor without parameters with a default directory (maybe not)(don't forget the Breaking change label if it is not the case)?

File: computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java [hunk 2]
```diff
@@ -65,4 +63,29 @@ public void log() {
...
+        Objects.requireNonNull(command);
+        if (index < 0) {
+            throw new IllegalArgumentException("Invalid index: " + index);
+        }
+
+        Path path = workingDirectory.resolve(command.getId() + "_" + index + extension);
+        try {
+            return Optional.of(Files.newInputStream(path));
+        } catch (IOException e) {
+            LOGGER.warn("Unable to read {}: {}", path.toString(), e.getMessage());
```
Comment (MioRtia) [inline computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java:86]: Sonarclound said it is not useful to add the `toString()` method as the logger already formats paths

File: computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java [hunk 3]
```diff
@@ -22,29 +27,22 @@
...
-    private static final ExecutionReport OK = new DefaultExecutionReport();
+    private final Path workingDirectory;
 
     private final List<ExecutionError> errors;
 
-    /**
-     * An execution report with no execution error.
-     * @return an execution report with no execution error.
-     */
-    public static ExecutionReport ok() {
```
Comment (MioRtia) [inline computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java:33]: Just for me, when did we use this method? (Don't forget the Breaking change label if we delete it)

File: computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java (hunk 2 above)
Comment (mathbagu) [reply computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java:86]: Fixed in the last commit

File: computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java (hunk 3 above)
Comment (mathbagu) [reply computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java:33]: This was used only in the unit test, but it's really useful to return always the same instance.

File: computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java (hunk 1 above)
Comment (mathbagu) [reply computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java:38]: Not sure it's relevant to have a default constructor. The working directory is the temporary directory used during the execution. From my point of view, computations always run in a temporary directory, even if no data are stored in it, it's created by the computation manager (part of the API).

Comment (MioRtia) [reply computation/src/main/java/com/powsybl/computation/DefaultExecutionReport.java:39]: Okay!

**Review by MioRtia — APPROVED** [review]

**Pull request merged**